src/runtime/finalizer.cs

```diff
@@ -87,24 +93,27 @@ internal static void Shutdown()
             }
             Instance.DisposeAll();
             Instance.CallPendingFinalizers();
-            Runtime.PyErr_Clear();
         }
 
         private void AddPendingCollect()
         {
+            if (_pending)
+            {
+                return;
+            }
```

Comment:
Is this optimisation really needed? If not, I'd rather like the code to be obviously safe.

Reply:
Just a simple double-check, make it check without lock at first.